src/pages/snippets/put-constraint-on-selector.mdx

```diff
@@ -0,0 +1,66 @@
+---
+title: Putting constraints on selectors
+description: "Making our components and directives more robust by putting constraints on their selectors"
+tags: ["angular"]
+pubDate: Mar 02, 2023
+contributedBy: "@Nartc1410"
+---
+
+`Component#selector` and `Directive#selector` are just like CSS Selector. This means we can utilize some CSS pseudo-class
+to put constraints on our `selector`, making them more robust and provide better DX to our consumers
+
+Suppose we have the following `NgxLilGui` component that wraps the `lil-gui` library.
+
+> To learn more about `lil-gui`, visit [https://lil-gui.georgealways.com/](https://lil-gui.georgealways.com/)
+
+```ts
+@Component({
+  selector: "ngx-lil-gui",
+  template: ` <ng-content></ng-content> `,
+  standalone: true,
+})
+export class NgxLilGui implements OnInit, OnDestroy {
+  @Input() config: Config;
+  @Input() object: Record<string, any>;
+}
```

Comment:
It's a tiny thing but, maybe it's worth adding `/* ... */` as you did in line 61. Cause now NgxLilGui misses ngOnInit and ngOnDestroy methods.